### PR TITLE
hwmv2: Port all arm xilinx socs and boards

### PR DESCRIPTION
**Pull request by nordicjm**
soc: xilinx_zyncmp: Port to HWMv2
    
    Ports the xilinx_zynqmp SoC configuration to hardware model
    version 2

    boards: arm: qemu_cortex_a9: Convert to v2
    
    Converts the board to hwmv2
    

    boards: arm: zybo: Convert to v2
    
    Converts the board to hwmv2

    soc: xilinx_zynq7000: Port to HWMv2
    
    Ports the xilinx_zynq7000 SoC configuration to hardware model
    version 2

- [x] needs v2 paths fixing, awaiting for that to be created on the branch

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]

**Review by ibirnbaum — APPROVED** [review]

File: boards/v2/qemu/qemu_cortex_a9/Kconfig.qemu_cortex_a9 [hunk 1]
```diff
@@ -6,6 +6,5 @@
 #
 
 config BOARD_QEMU_CORTEX_A9
-	bool "Cortex-A9 Emulation (QEMU)"
-	depends on SOC_XILINX_XC7Z007S
+	select SOC_XILINX_XC7Z007S
 	select QEMU_TARGET
```
Comment (tejlmand) [inline boards/v2/qemu/qemu_cortex_a9/Kconfig.qemu_cortex_a9:10]: the `select QEMU_TARGET` must either go to the `<board-dir>/Kconfig` or we must have a config qemu target in the Kconfig.soc tree.

File: soc/v2/xilinx/zynqmp/Kconfig.soc [hunk 2]
```diff
@@ -2,6 +2,9 @@
 # Copyright (c) 2019 Stephanos Ioannidis <root@stephanos.io>
 # SPDX-License-Identifier: Apache-2.0
 
+config SOC_XILINX_ZYNQMP
+	bool
+
 config SOC_XILINX_ZYNQMP_RPU
 	bool "Xilinx ZynqMP RPU"
 	select ARM
```
Comment (tejlmand) [inline soc/v2/xilinx/zynqmp/Kconfig.soc:10]: `select ARM` must go to the `xilinx/zynqmp/Kconfig` file.

File: soc/v2/xilinx/zynqmp/Kconfig.soc [hunk 3]
```diff
@@ -2,6 +2,9 @@
 # Copyright (c) 2019 Stephanos Ioannidis <root@stephanos.io>
 # SPDX-License-Identifier: Apache-2.0
 
+config SOC_XILINX_ZYNQMP
+	bool
+
 config SOC_XILINX_ZYNQMP_RPU
```
Comment (tejlmand) [inline soc/v2/xilinx/zynqmp/Kconfig.soc:8]: most of the selects done in this file must go to the `xilinx/zynqmp/Kconfig` file.

**Review by tejlmand — DISMISSED** [review]

File: boards/v2/qemu/qemu_cortex_a9/Kconfig.qemu_cortex_a9 (hunk 1 above)
Comment (nordicjm) [reply boards/v2/qemu/qemu_cortex_a9/Kconfig.qemu_cortex_a9]: Fixed

File: soc/v2/xilinx/zynqmp/Kconfig.soc (hunk 2 above)
Comment (nordicjm) [reply soc/v2/xilinx/zynqmp/Kconfig.soc]: Fixed

File: soc/v2/xilinx/zynqmp/Kconfig.soc (hunk 3 above)
Comment (nordicjm) [reply soc/v2/xilinx/zynqmp/Kconfig.soc]: Fixed

**Review by gmarull — APPROVED** [review]

**Pull request merged**